dev/tests/proxy.rs

```diff
@@ -143,6 +209,62 @@ async fn simple_backend_proxy() -> Result<()> {
     Ok(())
 }
 
+#[integration_test]
+async fn simple_ws_backend_proxy() -> Result<()> {
+    let proxy = Proxy::new().await?;
+    let server = Server::serve_web_sockets().await?;
+
+    let sr = base_spawn_request();
+    proxy.db.insert_backend(&sr).await?;
+
+    proxy
+        .db
+        .insert_proxy_route(&sr.backend_id, "foobar", &server.address.to_string())
+        .await?;
+
+    let (ws_stream, ws_handshake_resp) = proxy
+        .https_websocket("foobar", "/")
+        .await
+        .expect("could not request ws from proxy");
+
+    assert_eq!(
+        ws_handshake_resp.status(),
+        http::status::StatusCode::SWITCHING_PROTOCOLS
+    );
+
+    let (mut write, read): (
+        SplitSink<WebSocketStream<TlsStream<TcpStream>>, Message>,
```

Comment:
The stream types do not need to be specified here. I'm open to redundantly specifying types when it adds to readability, but I think in this case it's better to omit them.

This will also mean that some imports become unnecessary; `cargo test` will warn about them until they are cleaned up.